snuba/consumers/consumer.py

```diff
@@ -601,15 +605,69 @@ class MultistorageKafkaPayload(NamedTuple):
 ]
 
 
+def __message_to_dict(
+    message: Message[KafkaPayload],
+) -> Dict[str, Any]:
+    # TODO: Remove temporary mitigations for DLQ prod test
+    return {
+        "message_payload_value": message.payload.value,
+        "message_offset": message.offset,
+        "message_partition": message.partition.index,
+        "message_topic": message.partition.topic.name,
+    }
+
+
 def process_message(
     processor: MessageProcessor, message: Message[KafkaPayload]
 ) -> Union[None, BytesInsertBatch, ReplacementBatch]:
-    result = processor.process_message(
-        rapidjson.loads(message.payload.value),
-        KafkaMessageMetadata(
-            message.offset, message.partition.index, message.timestamp
-        ),
-    )
+
+    # TODO: Remove temporary mitigations for DLQ prod test
+
+    # Mitigation: Last resort - Hardcode message(s) to skip in runtime config
+    # expected format is "[topic:partition_index:offset,...]" eg [snuba-metrics:0:1,snuba-metrics:0:3]
+    messages_to_skip = (get_config("kafka_messages_to_skip") or "[]")[1:-1].split(",")
+    if (
+        f"{message.partition.topic.name}:{message.partition.index}:{message.offset}"
+        in messages_to_skip
+    ):
+        logger.warning(
+            f"A consumer for {message.partition.topic.name} skipped a message!",
+            exc_info=True,
+            extra=__message_to_dict(message),
+        )
+        return None
```

Comment:
you should write a test to make sure this is parsing the way you expect it to parse

Reply:
✅